lib/resolvers/error.js

```diff
@@ -31,14 +34,48 @@ const _cdsToGraphQLError = (context, err) => {
 }
 
 const _ensureError = error => (error instanceof Error ? error : new Error(error))
+const _clone = obj => Object.create(Object.getPrototypeOf(obj), Object.getOwnPropertyDescriptors(obj))
 
 const handleCDSError = (context, error) => {
   error = _ensureError(error)
-  // TODO: choose log level depending on type of error analogous to OData adapter
-  if (LOG_CDS._error) LOG_CDS.error(error)
+  _log(error)
   return _cdsToGraphQLError(context, error)
 }
 
+// TODO: Revise this logging functionality, as it's not specific to protocol adapters.
+// This function should be relocated and/or cleaned up when the new abstract/generic
+// protocol adapter is designed and implemented.
+const _log = error => {
+  // do not log standard errors, e.g. TypeError
+  if (error.message === undefined) return
+
```

Comment:
I think we should remove this check for now, since it doesn't seem to work correctly. As far as I know, all error objects will have the property `message` and all errors that are passed to this function are error objects since they are normalized to be so in `_ensureError`.

Also, not all errors that don't have the property `message` before normalization are non-standard errors. So moving the check before normalization wouldn't be a solution either.

Additionally, standard errors are also logged for OData requests. So I think we should just omit the check entirely.
```suggestion
```

Reply:
Yes, this check does not work reliably in all scenarios. Although in CAP, you can't create an error via the `cds.error()` interface that does not include the `message`, as the `message` is a required argument. For example:

```js
new cds.error()
```

This would result in the following uncaught `TypeError`:

```
Uncaught TypeError TypeError: Cannot read properties of undefined (reading 'raw')
```

Creating an error with an empty message via `new Error()` is possible. However, those errors contain the property `message` in the prototype chain assigned to an empty string; therefore, the check would not work as expected.